packages/ember-data/lib/system/model/errors.js

```diff
@@ -169,6 +181,41 @@ export default Ember.Object.extend(Ember.Enumerable, Ember.Evented, {
   messages: Ember.computed.mapBy('content', 'message'),
 
   /**
+    An array containing all of the error messages, capitalized and prefixed
+    with their attribute, for this record.
+
+    ```javascript
+    user.get('errors').add('firstName', 'is required');
+    user.get('errors').add('lastName', 'is required');
+
+    user.get('errors.fullMessages');
+    // ["First name is required", "Last name is required"]
+    ```
+
+    This is useful for displaying all errors to the user.
+
+    ```handlebars
+    {{#each model.errors.fullMessages as |message|}}
+      <div class="error">
+        {{message}}
+      </div>
+    {{/each}}
+    ```
+
+    @property fullMessages
+    @type {Array}
+  */
+  fullMessages: Ember.computed('content', function() {
+    var errors = get(this, 'content');
+
+    return map(errors, function(error) {
+      var attribute = underscore(error.attribute).replace('_', ' ');
```

Comment:
I'm not sure if there's a way to make this more flexible without having it use some function defined in your adapter.